docs/sections/papers/deita.md

```diff
@@ -0,0 +1,530 @@
+
+In distilabel 1.0, we initialise each step of the data generation
+pipeline. Later, we\'ll connect them together.
+
+
+```python
+evol_instruction_complexity = EvolInstruct(
+    name="evol_instruction_complexity",
+    llm=OpenAILLM(model="gpt-3.5-turbo"),
+    num_evolutions=5,
+    store_evolutions=True,
+    generate_answers=True,
+    include_original_instruction=True,
+    pipeline=pipeline,
+)
+
+evol_instruction_complexity.load()
+
+_evolved_instructions = next(evol_instruction_complexity.process(([{"instruction": "How many fish are there in a dozen fish?"}])))
+print(*_evolved_instructions, sep="\n")
+
+```
+Output:
+```bash
+( 1, 'How many fish are there in a dozen fish?')
+( 2, 'How many rainbow trout are there in a dozen rainbow trout?')
+( 3, 'What is the average weight in pounds of a dozen rainbow trout caught in a specific river in Alaska during the month of May?')
+```
+
+## 2. EVOL COMPLEXITY: Evaluate complexity of generated instructions
+
+The second step is the evaluation of *complexity* for an instruction in
+a given instruction-response pair. Like EVOL-INSTRUCT, this method
+leverages LLMs instead of humans to automatically score instructions.
+From any instruction-response pair, *(I, R)*, we first generate new
+instructions following the In-Depth Evolving Response. We generate more
+complex instructions through prompting, as explained by authors, by
+adding some constraints or reasoning steps. Let\'s take an example from
+[GPT-4-LLM](https://github.com/Instruction-Tuning-with-GPT-4/GPT-4-LLM)
+which aims to generate observations by GPT-4 to instruct-tune LLMs with
+supervised fine-tuning. And, we have the instruction *instruction\_0*:
+
+```python
+instruction_0 = "Give three tips for staying healthy."
+```
+
+To make it more complex, you can use, as the authors did, some prompt
+templates to add constraints or deepen the instruction. They provided
+some prompts in the paper appendix. For instance, this one was used to
+add constraints:
+```python
+PROMPT = """
+    I want you act as a Prompt Rewriter.
+    Your objective is to rewrite a given prompt into a more complex version to
+    make those famous AI systems (e.g., ChatGPT and GPT4) a bit harder to handle.
+    But the rewritten prompt must be reasonable and must be understood and
+    responded by humans.
+    Your rewriting cannot omit the non-text parts such as the table and code in
+    #Given Prompt#:. Also, please do not omit the input in #Given Prompt#.
+    You SHOULD complicate the given prompt using the following method:
+    Please add one more constraints/requirements into #Given Prompt#
+    You should try your best not to make the #Rewritten Prompt# become verbose,
+    #Rewritten Prompt# can only add 10 to 20 words into #Given Prompt#.
+    ‘#Given Prompt#’, ‘#Rewritten Prompt#’, ‘given prompt’ and ‘rewritten prompt’
+    are not allowed to appear in #Rewritten Prompt#
+    #Given Prompt#:
+    <Here is instruction>
+    #Rewritten Prompt#:
+    """
```

Comment:
Also note here that the tabs will be injected within the prompt, to ignore those you should either replace those or use `textwrap.dedent`